storage_service/src/main.rs

```diff
@@ -1,57 +1,403 @@
+                    if let Some(mt) = json_manifest_type {
+                        if manifest_type_param.is_none() {
+                            entry.manifest_type = mt;
+                        }
+                    }
+
+                    debug!("Using manifest_type: {}", entry.manifest_type);
+                    entry.manifest_json = Some(json_value);
+                }
+                Err(e) => {
+                    error!("Failed to parse JSON: {:?}", e);
+                    return HttpResponse::BadRequest().body(format!("Invalid JSON format: {}", e));
+                }
+            }
+        }
+        ContentFormat::CBOR => {
+            // Store as base64 encoded string
+            let encoded = general_purpose::STANDARD.encode(&bytes);
+            entry.manifest_cbor = Some(encoded);
+
+            // Try to decode CBOR to extract manifest_type if possible
+            match serde_cbor::from_slice::<serde_json::Value>(&bytes) {
+                Ok(cbor_value) => {
+                    let cbor_manifest_type = cbor_value
+                        .get("manifest_type")
+                        .and_then(|v| v.as_str())
+                        .map(|s| s.to_string());
+
+                    // Use the manifest_type from CBOR if found and not overridden by query parameter
+                    if let Some(mt) = cbor_manifest_type {
+                        if manifest_type_param.is_none() {
+                            entry.manifest_type = mt;
+                        }
+                    } else if manifest_type_param.is_none() {
+                        entry.manifest_type = "cbor_manifest".to_string();
+                    }
+                }
+                Err(e) => {
+                    debug!("Could not extract manifest_type from CBOR: {:?}", e);
+                    if manifest_type_param.is_none() {
+                        entry.manifest_type = "cbor_manifest".to_string();
+                    }
+                }
+            }
+        }
+        ContentFormat::Binary => {
+            // Store as base64 encoded string
+            let encoded = general_purpose::STANDARD.encode(&bytes);
+            entry.manifest_binary = Some(encoded);
+
+            // Use the manifest_type from query parameter or default
+            if manifest_type_param.is_none() {
+                entry.manifest_type = "binary_manifest".to_string();
+            }
+        }
+    }
+
     match collection.insert_one(entry, None).await {
         Ok(result) => {
-            info!("Successfully stored manifest with ID: {}", result.inserted_id);
+            info!(
+                "Successfully stored manifest with ID: {}",
+                result.inserted_id
+            );
+
+            // Add to Merkle Tree
+            let leaf = LogLeaf {
+                manifest_id: manifest_id,
+                hash,
+                sequence_number: sequence_number as u64,
+                timestamp: now,
```

Comment:
Similarly, do we want to include the sequence number and timestamp in the leaf hash? It's fairly common for Merkle trees to hash the entire contents of nodes, rather than just of some data fields, for the tamper evidence over the entire leaf.